File: Frontend-Dashboard/src/lib/level1ProgramCatalog.ts
/**
 * Stable Level 1 program slugs — sync with Backend/accounts/level1_program_catalog.py
 * After seed_syndicate_catalog, playlist IDs change; slugs stay fixed.
 */

export const LEVEL1_PSYCHOLOGY_SLUGS = [
  "level1-psych-01",
  "level1-psych-02",
  "level1-psych-03",
  "level1-psych-04",
  "level1-psych-05",
  "level1-psych-06",
  "level1-psych-07",
  "level1-psych-08",
  "level1-psych-09",
  "level1-psych-10",
  "level1-psych-11",
] as const;

export const LEVEL1_BUSINESS_MODEL_SLUGS = [
  "level1-model-01",
  "level1-model-02",
  "level1-model-03",
  "level1-model-04",
  "level1-model-05",
  "level1-model-06",
  "level1-model-07",
  "level1-model-08",
  "level1-model-09",
  "level1-model-10",
  "level1-model-11",
] as const;

export const PUBLIC_LEVEL1_PLAYLIST_SLUGS = new Set<string>([
  ...LEVEL1_PSYCHOLOGY_SLUGS,
  ...LEVEL1_BUSINESS_MODEL_SLUGS,
]);

/** Display order on /programs (Business Psychology column) — all 11. */
export const PUBLIC_PSYCHOLOGY_SLUG_ORDER: readonly string[] = [
  "level1-psych-03",
  "level1-psych-04",
  "level1-psych-08",
  "level1-psych-07",
  "level1-psych-09",
  "level1-psych-10",
  "level1-psych-11",
  "level1-psych-01",
  "level1-psych-06",
  "level1-psych-02",
  "level1-psych-05",
];

/** Canonical titles — sync with Backend/accounts/level1_program_catalog.py */
export const LEVEL1_CANONICAL_TITLES: Record<string, string> = {
  "level1-psych-01": "The 9 to 5 Exit Strategy",
  "level1-psych-02": "Zero to One Million",
  "level1-psych-03": "Hustle Hard",
  "level1-psych-04": "Mastering Consistency",
  "level1-psych-05": "The Secret To Transformation",
  "level1-psych-06": "The Compound Effect",
  "level1-psych-07": "The Micro Business Protocol",
  "level1-psych-08": "Mastering Risk and Uncertainty",
  "level1-psych-09": "Business Warfare",
  "level1-psych-10": "Syndicate 13 Business Rules",
  "level1-psych-11": "Syndicate Money Philosophy",
  "level1-model-01": "N8N AI Automation",
  "level1-model-02": "AI Automations",
  "level1-model-03": "App Building (using Flutter)",
  "level1-model-04": "Building Apps using React JS",
  "level1-model-05": "Book Publishing On Amazon (KINDLE)",
  "level1-model-06": "Building Games Using Unreal Engine",
  "level1-model-07": "Framer Crash Course",
  "level1-model-08": "Graphics Design Using Canva",
  "level1-model-09": "Print On Demand Clothing",
  "level1-model-10": "Python Programming",
  "level1-model-11": "WordPress Blog",
};

/**
 * Legacy stream-playlist-catalog.json ids → stable level1 slug (deep links / globe tiles).
 * API playlist ids after seed may differ; slug is the canonical key.
 */
export const LEGACY_PROGRAM_ID_TO_LEVEL1_SLUG: Record<number, string> = {
  1: "level1-psych-01",
  2: "level1-psych-02",
  3: "level1-psych-03",
  6: "level1-psych-04",
  9: "level1-psych-05",
  12: "level1-psych-06",
  31: "level1-psych-07",
  30: "level1-psych-08",
  99: "level1-psych-09",
  7: "level1-psych-10",
  8: "level1-psych-11",
  17: "level1-model-01",
  16: "level1-model-02",
  21: "level1-model-03",
  28: "level1-model-04",
  25: "level1-model-05",
  20: "level1-model-06",
  14: "level1-model-07",
  23: "level1-model-08",
  19: "level1-model-09",
  24: "level1-model-10",
  13: "level1-model-11",
};

/** Display order on /programs (Business Model column). */
export const PUBLIC_BUSINESS_MODEL_SLUG_ORDER: readonly string[] = [
  "level1-model-01",
  "level1-model-02",
  "level1-model-03",
  "level1-model-04",
  "level1-model-05",
  "level1-model-06",
  "level1-model-07",
  "level1-model-08",
  "level1-model-09",
  "level1-model-10",
  "level1-model-11",
];

export const LEVEL1_SLUG_DISPLAY_ORDER: readonly string[] = [
  ...PUBLIC_PSYCHOLOGY_SLUG_ORDER,
  ...PUBLIC_BUSINESS_MODEL_SLUG_ORDER,
];

export const LEVEL1_PSYCHOLOGY_PROGRAM_TITLES: readonly string[] = LEVEL1_PSYCHOLOGY_SLUGS.map(
  (slug) => LEVEL1_CANONICAL_TITLES[slug] ?? slug
);

export const LEVEL1_BUSINESS_MODEL_PROGRAM_TITLES: readonly string[] = LEVEL1_BUSINESS_MODEL_SLUGS.map(
  (slug) => LEVEL1_CANONICAL_TITLES[slug] ?? slug
);

export const LEVEL1_SLUG_THUMBNAILS: Record<string, string> = {
  "level1-psych-01": "/assets/programs/cources%20imnages/9-5.jpg",
  "level1-psych-02": "/assets/programs/cources%20imnages/0%20to%201M.jpg",
  "level1-psych-03": "/assets/programs/cources%20imnages/hustle.jpg",
  "level1-psych-04": "/assets/programs/cources%20imnages/consistency.jpg",
  "level1-psych-05": "/assets/programs/cources%20imnages/secret.jpg",
  "level1-psych-06": "/assets/programs/cources%20imnages/compound%20effect.jpg",
  "level1-psych-07": "/assets/programs/cources%20imnages/micro%20business.jpg",
  "level1-psych-08": "/assets/programs/cources%20imnages/uncertainty.jpg",
  "level1-psych-09": "/assets/programs/cources%20imnages/warfare.jpg",
  "level1-psych-10": "/assets/programs/cources%20imnages/13rules.jpg",
  "level1-psych-11": "/assets/programs/cources%20imnages/money-philosophy.jpg",
  "level1-model-01": "/assets/programs/cources%20imnages/business-content-n8n-ai-automation.png",
  "level1-model-02": "/assets/programs/cources%20imnages/ai-content-automation-business.png",
  "level1-model-03": "/assets/programs/cources%20imnages/app-building-vibe-coding.png",
  "level1-model-04": "/assets/programs/cources%20imnages/custom-app-blueprint.png",
  "level1-model-05": "/assets/programs/cources%20imnages/ebook-business-blueprint.png",
  "level1-model-06": "/assets/programs/cources%20imnages/gaming-business-blueprint.png",
  "level1-model-07": "/assets/programs/cources%20imnages/rapid-noncode-web-building.png",
  "level1-model-08": "/assets/programs/cources%20imnages/graphics-design-for-business.png",
  "level1-model-09": "/assets/programs/cources%20imnages/zero-inventory-clothing-blueprint.png",
  "level1-model-10": "/assets/programs/cources%20imnages/basics-python-small-business.png",
  "level1-model-11": "/assets/programs/cources%20imnages/profitable-blogging-blueprint.png",
};

export const LEVEL1_SLUG_TITLE_OVERRIDES: Record<string, string> = {
  "level1-model-01": "AI content Automation for Businesses",
  "level1-model-02": "Social Media Content Automation",
  "level1-model-03": "App Building for Business (Vibe Coding)",
  "level1-model-04": "The Custom App Blueprint for Business",
  "level1-model-05": "eBook Business Blueprint (Monetize Your Knowledge)",
  "level1-model-06": "The Gaming Business Blueprint (Build, Launch, and Sell)",
  "level1-model-07": "Rapid Non-code Web Building (For Business)",
  "level1-model-08": "Graphics Design for Business (Graphics That Convert to Sales)",
  "level1-model-09": "The Zero-Inventory Clothing Business Blueprint",
  "level1-model-10": "Basics Python for Small Business",
  "level1-model-11": "The Profitable Blogging Blueprint",
  "level1-psych-07": "Micro Business Protocols",
};

File: Backend/accounts/level1_program_catalog.py
"""
Level 1 Programs — 11 Business Psychology + 11 Business Model playlists.

Stable catalog slugs (level1-psych-*, level1-model-*) survive DB wipes.
R2 layout: ``Business Psychology/{outer}/…/index.m3u8`` and ``Business Models/{outer}/…/index.m3u8``.
"""

from __future__ import annotations

from dataclasses import dataclass
from decimal import Decimal

from apps.video_streaming.models import StreamPlaylist

LEVEL1_BUSINESS_PSYCHOLOGY_UNIT_USD = Decimal("99.00")
LEVEL1_BUSINESS_MODEL_UNIT_USD = Decimal("75.00")

# Deprecated: prefer per-program unit prices. Kept for older imports.
LEVEL1_CATEGORY_TOTAL_USD = LEVEL1_BUSINESS_MODEL_UNIT_USD * 11


@dataclass(frozen=True)
class Level1ProgramRow:
    catalog_slug: str
    title: str
    category: str
    r2_root: str
    r2_outer_folder: str
    r2_lesson_folder: str
    display_order: int


def _psych(
    index: int,
    title: str,
    outer: str,
    lesson: str,
) -> Level1ProgramRow:
    return Level1ProgramRow(
        catalog_slug=f"level1-psych-{index:02d}",
        title=title,
        category=StreamPlaylist.Category.BUSINESS_PSYCHOLOGY,
        r2_root="Business Psychology",
        r2_outer_folder=outer,
        r2_lesson_folder=lesson,
        display_order=index,
    )


def _model(
    index: int,
    title: str,
    outer: str,
    lesson: str,
) -> Level1ProgramRow:
    return Level1ProgramRow(
        catalog_slug=f"level1-model-{index:02d}",
        title=title,
        category=StreamPlaylist.Category.BUSINESS_MODEL,
        r2_root="Business Models",
        r2_outer_folder=outer,
        r2_lesson_folder=lesson,
        display_order=index,
    )


LEVEL1_PSYCHOLOGY_PROGRAMS: tuple[Level1ProgramRow, ...] = (
    _psych(1, "The 9 to 5 Exit Strategy", "9 to 5 Exit", "Lesson 1 - 9 to 5 Exit Strategy"),
    _psych(2, "Zero to One Million", "Zero to 1 Million", "Lesson 1 - Zero to One Million"),
    _psych(3, "Hustle Hard", "Hustle Hard", "Intro"),
    _psych(4, "Mastering Consistency", "Mastering Consistency", "Intro"),
    _psych(5, "The Secret To Transformation", "Secret To Transformation", "Intro"),
    _psych(6, "The Compound Effect", "Compound Effect", "Intro"),
    _psych(7, "The Micro Business Protocol", "Micro Business Protocol", "Intro"),
    _psych(8, "Mastering Risk and Uncertainty", "Risk and Uncertainty", "Intro"),
    _psych(9, "Business Warfare", "Business Warfare", "Intro"),
    _psych(10, "Syndicate 13 Business Rules", "13 Business Rules", "Intro"),
    _psych(11, "Syndicate Money Philosophy", "Money Philosophy", "Intro"),
)

LEVEL1_BUSINESS_MODEL_PROGRAMS: tuple[Level1ProgramRow, ...] = (
    _model(1, "N8N AI Automation", "N8N AI Automation", "Intro"),
    _model(2, "AI Automations", "Ai Automation", "Intro"),
    _model(3, "App Building (using Flutter)", "App Building Flutter", "Intro"),
    _model(4, "Building Apps using React JS", "React", ""),
    _model(5, "Book Publishing On Amazon (KINDLE)", "Amazon KDP", ""),
    _model(6, "Building Games Using Unreal Engine", "Unreal Engine", ""),
    _model(7, "Framer Crash Course", "Framer Crash Course", "Intro"),
    _model(8, "Graphics Design Using Canva", "Canva", "Intro"),
    _model(9, "Print On Demand Clothing", "Print On Demand", "Intro"),
    _model(10, "Python Programming", "Python", ""),
    _model(11, "WordPress Blog", "WordPress Blog", "Intro"),
)

LEVEL1_ALL_PROGRAMS: tuple[Level1ProgramRow, ...] = LEVEL1_PSYCHOLOGY_PROGRAMS + LEVEL1_BUSINESS_MODEL_PROGRAMS

LEVEL1_CATALOG_SLUGS: frozenset[str] = frozenset(row.catalog_slug for row in LEVEL1_ALL_PROGRAMS)

KNIGHT_SELECTABLE_LEVEL1_SLUGS: frozenset[str] = frozenset(
    list(row.catalog_slug for row in LEVEL1_PSYCHOLOGY_PROGRAMS[:9])
    + list(row.catalog_slug for row in LEVEL1_BUSINESS_MODEL_PROGRAMS)
)
